src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1328 @@
+    {
+        // switch to default window..
+        $this->switchToWindow();
+        // ..and close all other windows
+        foreach ($this->getWindowNames() as $name) {
+            if ($name !== $this->initialWindowName) {
+                $this->withWindow($name, fn() => $this->getWebDriver()->close());
+            }
+        }
+
+        $this->getWebDriver()->manage()->deleteAllCookies();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visit($url): void
+    {
+        $this->getWebDriver()->navigate()->to($url);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentUrl(): string
+    {
+        return $this->getWebDriver()->getCurrentURL();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reload(): void
+    {
+        $this->getWebDriver()->navigate()->refresh();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function forward(): void
+    {
+        $this->getWebDriver()->navigate()->forward();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function back(): void
+    {
+        $this->getWebDriver()->navigate()->back();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function switchToWindow($name = null): void
+    {
+        if ($name === null) {
+            $name = $this->initialWindowName;
+        }
+
+        if (is_string($name)) {
+            $name = $this->getWindowHandleFromName($name);
+        }
+
+        $this->getWebDriver()->switchTo()->window((string)$name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function switchToIFrame($name = null): void
+    {
+        $frameQuery = $name;
+        if ($name && $this->getWebDriver()->isW3cCompliant()) {
+            try {
+                $frameQuery = $this->getWebDriver()->findElement(WebDriverBy::id($name));
+            } catch (NoSuchElementException $e) {
+                $frameQuery = $this->getWebDriver()->findElement(WebDriverBy::name($name));
+            }
+        }
+
+        $this->getWebDriver()->switchTo()->frame($frameQuery);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCookie($name, $value = null): void
+    {
+        if (null === $value) {
+            $this->getWebDriver()->manage()->deleteCookieNamed($name);
+
+            return;
+        }
+
+        $cookieArray = [
+            'name' => $name,
+            'value' => rawurlencode($value),
+            'secure' => false,
+        ];
+
+        $this->getWebDriver()->manage()->addCookie($cookieArray);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCookie($name): ?string
+    {
+        try {
+            $result = $this->getWebDriver()->manage()->getCookieNamed($name);
+        } catch (NoSuchCookieException $e) {
+            $result = null;
+        }
+        if ($result === null) {
+            return null;
+        }
+
+        $result = $result->getValue();
+        if ($result === null) {
+            return null;
```

Comment:
this condition looks like dead code. The php-webdriver defines the return type of `getValue` as being `string`